src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -85,15 +83,43 @@ def execute(self, unexpanded_argv):
         self.raise_event(self.COMMAND_PARSER_LOADED, parser=self.parser)
 
         if len(argv) == 0:
+            enable_autocomplete(self.parser)
             az_subparser = self.parser.subparsers[tuple()]
             _help.show_welcome(az_subparser)
             log_telemetry('welcome')
             return None
 
         if argv[0].lower() == 'help':
             argv[0] = '--help'
+        
+        # Rudimentary parsing to get the command
+        nouns = []
+        for noun in argv:
+            if noun[0] == '-':
+                break
+            nouns.append(noun)
+        command = ' '.join(nouns)
+
+        if argv[-1] in ('--help', '-h'):
+            # Make sure we load the params so they show in help
+            self.configuration.load_params(command)
+            self.parser.load_command_table(command_table)
+
+        if self.session['completer_active']:
+            # Make sure we load the params for tab completion then enable it
+            self.configuration.load_params(command)
+            self.parser.load_command_table(command_table)
+            enable_autocomplete(self.parser)
+
+        # Parse once so we can get the command
```

Comment:
The check at 101 is rudimentary and doesn't do everything that argparse does.
If `self.parser.parse_args()` returns, we know it parsed successfully so we can get the command from argparse.